graders/c/Dockerfile

```diff
@@ -11,20 +11,28 @@ ENV PYTHONIOENCODING=UTF-8
 ENV PYTHONPATH=/cgrader/:/grade/serverFilesCourse
 ENV C_INCLUDE_PATH=/cgrader
 
+# - Ensure that all `uv` commands compile Python source files to bytecode.
+# - Ensure that all `uv` commands do not use any caching.
+ENV UV_COMPILE_BYTECODE=1 UV_NO_CACHE=1
+
 RUN apt-get update \
-    && apt-get install -y --no-install-recommends gcc g++ gdb make valgrind check pkg-config clang python3.12 python3-pip \
+    && apt-get install -y --no-install-recommends gcc g++ gdb make valgrind check pkg-config clang curl ca-certificates \
     && apt-get clean \
     && rm -rf /var/lib/apt/lists/* \
-    && ln -sf /usr/bin/python3.12 /usr/bin/python3 \
     # The clang Python package typically expects to see libclang.so in /usr/lib,
     # which would be the case if we installed libclang-dev, but since that
     # package is too large and mostly unnecessary, we add a symlink to the
     # existing libclang.so.1.
     && ln -s /usr/lib/llvm-18/lib/libclang.so.1 /usr/lib/libclang.so \
     && groupadd sbuser && useradd -g sbuser sbuser
 
+# Install uv and Python
+RUN curl -LO https://astral.sh/uv/install.sh \
+    && env UV_INSTALL_DIR=/usr/local/bin sh /install.sh && rm /install.sh \
+    && uv python install python3.12
+
 COPY requirements.txt /requirements.txt
-RUN pip3 install --no-cache-dir --break-system-packages -r /requirements.txt
+RUN uv pip install --system --break-system-packages -r /requirements.txt
```

Comment:
Using the system Python works, but we might be better off using a venv as in https://github.com/PrairieLearn/PrairieLearn/pull/13520?